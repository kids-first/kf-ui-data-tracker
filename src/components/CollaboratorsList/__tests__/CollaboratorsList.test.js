import React from 'react';
import {render, act, fireEvent} from '@testing-library/react';
import {CollaboratorsList} from '../';

jest.mock('auth0-js');

const users = [
  {
    node: {
      id: 'VXNlck5vZGU6OQ==',
      username: 'Anthony Hamilton',
      firstName: 'John',
      lastName: 'Garza',
      email: 'richard06@merritt.net',
      picture: '',
      roles: [],
      __typename: 'UserNode',
    },
    __typename: 'UserNodeEdge',
  },
  {
    node: {
      id: 'VXNlck5vZGU6MjY=',
      username: 'devadmin',
      firstName: '',
      lastName: '',
      email: 'admin@myproject.com',
      picture: '',
      roles: ['ADMIN'],
      __typename: 'UserNode',
    },
    __typename: 'UserNodeEdge',
  },
];

it('Renders collaborators list correctly for admins', async () => {
  const removeCollaborator = jest.fn();

  const tree = render(
    <CollaboratorsList
      users={users}
      showAdminActions={true}
      removeCollaborator={removeCollaborator}
    />,
  );
  expect(tree.container).toMatchSnapshot();
});

it('Renders collaborators list correctly for non-admins', async () => {
  const removeCollaborator = jest.fn();

  const tree = render(
    <CollaboratorsList
      users={users}
      showAdminActions={false}
      removeCollaborator={removeCollaborator}
    />,
  );
  expect(tree.container).toMatchSnapshot();
});

it('Calls remove collaborator mutation', async () => {
  const removeCollaborator = jest.fn();

  const tree = render(
    <CollaboratorsList
      users={users}
      showAdminActions={true}
      removeCollaborator={removeCollaborator}
    />,
  );

  // Start with two collaborators
  expect(tree.queryAllByTestId('remove-button').length).toBe(2);

  act(() => {
    fireEvent.click(tree.queryAllByTestId('remove-button')[0]);
  });

  act(() => {
    fireEvent.click(tree.queryAllByTestId('remove-confirm')[0]);
  });

  // End with one collaborator
  expect(removeCollaborator.mock.calls.length).toBe(1);
});
