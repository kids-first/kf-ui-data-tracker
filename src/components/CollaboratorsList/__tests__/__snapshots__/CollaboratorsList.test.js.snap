// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders collaborators list correctly for admins 1`] = `
<div>
  <div
    class="ui divided relaxed list"
    role="list"
  >
    <div
      class="item"
      data-testid="user-item"
      role="listitem"
    >
      <img
        alt="John Garza"
        class="ui avatar image"
        src="test-file-stub"
      />
      <div
        class="content"
      >
        <div
          class="header"
        >
          John Garza
          <small>
             - 
            richard06@merritt.net
          </small>
        </div>
        <div
          class="description"
        >
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              Joined
               
            </div>
            <div
              class="item"
              role="listitem"
            >
              Added by 
              Unknown user
            </div>
          </div>
        </div>
      </div>
      <div
        class="right floated content"
      >
        <div
          class="ui mini fluid buttons"
        >
          <button
            class="ui basic icon negative left labeled button"
            data-testid="remove-button"
          >
            <i
              aria-hidden="true"
              class="remove icon"
            />
            Remove
          </button>
        </div>
      </div>
    </div>
    <div
      class="item"
      data-testid="user-item"
      role="listitem"
    >
      <img
        alt="devadmin"
        class="ui avatar image"
        src="test-file-stub"
      />
      <div
        class="content"
      >
        <div
          class="header"
        >
          devadmin
          <small>
             - 
            admin@myproject.com
          </small>
        </div>
        <div
          class="description"
        >
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              Joined
               
            </div>
            <div
              class="item"
              role="listitem"
            >
              Added by 
              Unknown user
            </div>
          </div>
        </div>
      </div>
      <div
        class="right floated content"
      >
        <div
          class="ui mini fluid buttons"
        >
          <button
            class="ui basic icon negative left labeled button"
            data-testid="remove-button"
          >
            <i
              aria-hidden="true"
              class="remove icon"
            />
            Remove
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders collaborators list correctly for non-admins 1`] = `
<div>
  <div
    class="ui divided relaxed list"
    role="list"
  >
    <div
      class="item"
      data-testid="user-item"
      role="listitem"
    >
      <img
        alt="John Garza"
        class="ui avatar image"
        src="test-file-stub"
      />
      <div
        class="content"
      >
        <div
          class="header"
        >
          John Garza
          <small>
             - 
            richard06@merritt.net
          </small>
        </div>
        <div
          class="description"
        >
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              Joined
               
            </div>
            <div
              class="item"
              role="listitem"
            >
              Added by 
              Unknown user
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="item"
      data-testid="user-item"
      role="listitem"
    >
      <img
        alt="devadmin"
        class="ui avatar image"
        src="test-file-stub"
      />
      <div
        class="content"
      >
        <div
          class="header"
        >
          devadmin
          <small>
             - 
            admin@myproject.com
          </small>
        </div>
        <div
          class="description"
        >
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              Joined
               
            </div>
            <div
              class="item"
              role="listitem"
            >
              Added by 
              Unknown user
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
